pkg/cmd/healthprobe/healthprobe.go

```diff
@@ -0,0 +1,316 @@
+package healthprobe
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/openshift/library-go/pkg/serviceability"
+	"io"
+	"io/ioutil"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"go.etcd.io/etcd/pkg/transport"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DefaultDialTimeout         = 2 * time.Second
+	DefaultHealthCheckInterval = 5 * time.Second
+	DefaultEndpoint            = "https://localhost:2379"
```

Comment:
our `etcdctl` container has env vars that I thought enabled it to check all the members.

Reply:
So to be clear this is a quorate get. The current test 
- checks etcd is available (localhost)
- checks that etcd has quorum so N+1 etcds are available thus checking host networking is available.

If this is done on each node that is pretty reasonable coverage with 3 RPS. If each node is checking each member the cost will increase 9 RPS. I didn't think that was required but I am open to better understanding why you feel that is required. KAS needs a single subconn in client balancer to be `Ready` and the cluster to quorate.

Reply:
I can see now your point with local etcd being down but cluster quorate from KAS point of view.